### PR TITLE
feat: implement frontend-local admin API routes with SQLite storage

### LINKED ISSUE #42: Implement frontend-local admin API routes with SQLite storage
## Context

The admin panel UI is fully built (`/admin/content`, `/admin/images`, `/admin/layout-settings`) but currently calls backend API endpoints (`NEXT_PUBLIC_API_BASE/api/admin/*`) that don't exist. These admin settings are **frontend presentation concerns** — layout config, Tailwind classes, page content, image display defaults — and should live in the frontend, not the backend.

The frontend already has a SQLite database (`auth.db` via BetterAuth + `better-sqlite3`) and its own API routes (`/api/auth/[...all]`, `/api/checkout`). The admin endpoints should follow the same pattern.

## Architecture Change

**Before:** Frontend admin → cross-origin fetch to backend → MongoDB (endpoints never built)
**After:** Frontend admin → same-origin fetch to frontend API routes → SQLite

This eliminates:
- Cross-origin complexity (CORS, credentials, port mismatches)
- Dependency on backend for admin functionality
- The need for the `/api/:path*` rewrite proxy for admin routes

The backend remains focused on domain data: photos, categories, galleries, orders, Stripe.

### 1. Extend SQLite schema

Add tables to the existing `auth.db` (or a separate `admin.db` if preferred):

```sql
CREATE TABLE IF NOT EXISTS pages (
  id TEXT PRIMARY KEY,
  title TEXT NOT NULL,
  content TEXT DEFAULT '',
  createdAt TEXT DEFAULT (datetime('now')),
  updatedAt TEXT DEFAULT (datetime('now'))
);

CREATE TABLE IF NOT EXISTS settings (
  key TEXT PRIMARY KEY,
  value TEXT NOT NULL,
  updatedAt TEXT DEFAULT (datetime('now'))
);

CREATE TABLE IF NOT EXISTS photo_image_settings (
  photoId TEXT PRIMARY KEY,
  settings TEXT NOT NULL,
  updatedAt TEXT DEFAULT (datetime('now'))
);
```

The `settings` table stores JSON blobs keyed by name (`"images"` for global image defaults, `"layout"` for layout config). This is simpler than separate tables for each setting type.

### 2. Create a shared database helper

**New file:** `lib/admin-db.js`

Initializes the SQLite connection and runs migrations (create tables if not exist). Imported by each API route handler.

### 3. Add API route handlers

All routes should validate the BetterAuth session before processing (reject 401 if unauthenticated).

| Route file | Methods | Purpose |
|------------|---------|---------|
| `app/api/admin/pages/route.js` | GET | List all pages |
| `app/api/admin/pages/[pageId]/route.js` | GET, PUT | Get/update single page |
| `app/api/admin/settings/images/route.js` | GET, PUT | Global image defaults |
| `app/api/admin/settings/layout/route.js` | GET, PUT | Layout configuration |
| `app/api/admin/photos/[photoId]/image-settings/route.js` | GET, PUT | Per-photo overrides |

**Session validation pattern:**
```js
import { auth } from '@/lib/auth'
import { headers } from 'next/headers'

export async function GET(request) {
  const session = await auth.api.getSession({ headers: await headers() })
  if (!session) return Response.json({ error: 'Unauthorized' }, { status: 401 })
  // ... handle request
}
```

### 4. Update frontend API consumers

**`lib/contentApi.js`** — Change from:
```js
const res = await fetch(`${base}/api/admin/pages`, { credentials: 'include' })
```
To:
```js
const res = await fetch('/api/admin/pages', { credentials: 'include' })
```

Same-origin calls don't need `NEXT_PUBLIC_API_BASE`. Update all 6 functions in the hook and 6 standalone exports.

**`lib/api.js`** — Same change for `getPhotoImageSettings()` and `updatePhotoImageSettings()`.

### 5. Seed default data

GET endpoints should return sensible defaults when no data exists (not 404):
- `GET /api/admin/pages` → `[]` (empty array)
- `GET /api/admin/settings/images` → `{}` (empty object, frontend uses its own defaults)
- `GET /api/admin/settings/layout` → default layout config:
  ```json
  {
    "showHeader": true,
    "showFooter": true,
    "gridColumns": 3,
    "colorScheme": "light",
    "navigationItems": [],
    "componentStyles": {}
  }
  ```
- `GET /api/admin/photos/:id/image-settings` → `{}` (falls back to global defaults in `ContentContext.js`)

### 6. Clean up rewrite proxy

The `next.config.mjs` rewrite was already changed to `fallback` during the demo to fix the BetterAuth conflict. With admin routes now local, the proxy is only needed for public data endpoints (`/api/categories`, `/api/photos`, etc.) that still come from the backend. Verify the fallback approach works for those, or remove the proxy entirely since `lib/api.js` already uses `NEXT_PUBLIC_API_BASE` directly.

## Files to modify

- `lib/admin-db.js` — **new** — SQLite helper with migrations
- `app/api/admin/pages/route.js` — **new** — pages list endpoint
- `app/api/admin/pages/[pageId]/route.js` — **new** — single page endpoint
- `app/api/admin/settings/images/route.js` — **new** — image settings endpoint
- `app/api/admin/settings/layout/route.js` — **new** — layout settings endpoint
- `app/api/admin/photos/[photoId]/image-settings/route.js` — **new** — per-photo settings
- `lib/contentApi.js` — change API base to relative URLs
- `lib/api.js` — change photo settings API base to relative URLs
- `next.config.mjs` — verify/simplify rewrite proxy

## Also found during demo (fix alongside)

- `better-sqlite3` native module needs rebuild for Node 24 (`npm rebuild better-sqlite3`)
- No admin user seed script exists — add one or document the process
- `next.config.mjs` rewrite was already changed from array to `fallback` format — commit this

## Acceptance criteria

- [ ] All admin pages load without API errors when backend is stopped
- [ ] Settings persist across page refreshes (stored in SQLite)
- [ ] Layout editor can save/revert component styles
- [ ] Content editor can list, view, and update pages
- [ ] Image settings page can read/write global defaults
- [ ] Per-photo overrides work (GET returns empty object when none set)
- [ ] Unauthenticated requests to admin routes return 401
- [ ] Existing public API calls to the backend still work

### PR DESCRIPTION
**Pull request by frankbria**
## Summary
Implements #42: Move admin settings from backend API dependency to local Next.js API routes backed by SQLite.

- Added `lib/admin-db.js` — lazy-initialized SQLite database helper with pages, settings, and photo_image_settings tables
- Created 5 admin API route handlers (`/api/admin/pages`, `/api/admin/pages/[pageId]`, `/api/admin/settings/images`, `/api/admin/settings/layout`, `/api/admin/photos/[photoId]/image-settings`) with BetterAuth session validation
- Updated `contentApi.js` and `api.js` to use relative URLs for admin endpoints (no longer depends on `NEXT_PUBLIC_API_BASE` for admin operations)
- Fixed `next.config.mjs` — corrected rewrite port (4000→4010) and switched to `fallback` rewrites to avoid intercepting BetterAuth routes

## Acceptance Criteria
- [x] Admin settings stored in frontend-local SQLite (`admin.db`)
- [x] All admin API routes require BetterAuth session validation
- [x] Admin API consumers use relative URLs (served by same Next.js app)
- [x] Public-facing endpoints (gallery, categories) still hit backend API
- [x] Lazy DB initialization prevents build-time SQLite locking
- [x] Existing tests updated, new tests added for all routes

## Test Plan
- [x] 708 tests passing (50 test suites)
- [x] ESLint: No warnings or errors
- [x] Production build: Successful (all admin routes marked as dynamic)
- [x] New test coverage: admin-db CRUD, all 5 route handlers (auth + happy path + edge cases), contentApi relative URLs, api.js relative URLs

Closes #42

## Summary by CodeRabbit

* **New Features**
  * Admin API for managing pages, per-photo image settings, and layout/image defaults
  * Local SQLite-backed admin datastore for persistent admin data

* **Tests**
  * Comprehensive unit tests for admin APIs and admin datastore covering auth, validation, CRUD, and error cases

* **Chores**
  * Internal API calls simplified to relative endpoints; development proxy port adjusted; gitignore expanded for DB artifacts

* **Documentation**
  * Updated env/example and developer guides to reflect new proxy port and routing behavior

### CODE REVIEW

**Comment by frankbria** [comment]
## Review Feedback Response

### Critical / Major — Accepted & Fixed

| # | Source | Issue | Action |
|---|--------|-------|--------|
| 1 | GitHub Actions (opencode) | **Missing JSON.parse error handling** in 3 settings GET routes | **Fixed** — added try-catch around `JSON.parse` in settings/images, settings/layout, and photos/image-settings routes. Returns sensible defaults (empty object or DEFAULT_LAYOUT) on parse failure instead of 500. |
| 2 | GitHub Actions (opencode) | **Missing `export const dynamic = 'force-dynamic'`** on all admin routes | **Fixed** — added to all 5 route files. Without this, Next.js could cache GET responses and serve stale admin data. |
| 3 | Macroscope | **Layout settings stored without schema validation** | **Fixed** (prior commit) — validates `gridColumns` (1-12), `colorScheme` (light/dark), and object type. |
| 4 | CodeRabbit | **Port mismatch**: next.config.mjs uses 4010 but docs say 4000 | **Fixed** (prior commit) — updated CLAUDE.md, AGENTS.md, .env.example to 4010. |

### Major — Rejected with Reasoning

| # | Source | Issue | Decision |
|---|--------|-------|----------|
| 5 | CodeRabbit | **Relative URLs conflict with `NEXT_PUBLIC_API_BASE` guideline** | **Rejected.** This is a false positive. The entire purpose of issue #42 is to move admin endpoints from the backend to local Next.js API routes backed by SQLite. Using `NEXT_PUBLIC_API_BASE` for local routes would incorrectly proxy them to the backend. Relative URLs are the correct architecture — the `fallback` rewrite pattern in `next.config.mjs` ensures local routes (auth, admin) take precedence while unmatched requests proxy to the backend. CLAUDE.md has been updated to document this distinction. |

### Minor — Accepted & Fixed

| # | Source | Issue | Action |
|---|--------|-------|--------|
| 6 | CodeRabbit | **`global.Response` mock not restored in test teardown** | **Fixed** — all 5 admin route test files now save `originalResponse` and restore in `afterAll()`. |
| 7 | GitHub Actions | **Inconsistent input validation** across PUT routes | **Fixed** — added object-type validation to images and photo-settings PUT routes for consistency with page and layout routes. |

### Minor — Deferred / Rejected

| # | Source | Issue | Decision |
|---|--------|-------|----------|
| 8 | GitHub Actions | **Duplicate auth code** — extract to helper | **Deferred.** 5 small route files with identical 3-line auth checks don't warrant a shared abstraction yet. Extracting now adds indirection without meaningful DRY benefit. Will revisit if more admin routes are added. |
| 9 | GitHub Actions | **Missing DELETE operations** for pages | **Out of scope.** Issue #42 specifies GET/PUT for pages. DELETE can be added in a follow-up issue if needed. |
| 10 | GitHub Actions | **Payload size limits** | **Not needed.** Next.js enforces a default 4MB body size limit on API routes. Explicit validation adds no value here. |
| 11 | GitHub Actions | **No database indexes** | **Premature.** Admin settings tables have single-digit rows. Indexes would add overhead for zero benefit at this scale. |
| 12 | GitHub Actions | **`jest.resetModules()` is heavy-handed** in admin-db tests | **Intentional.** The tests need a fresh `:memory:` SQLite database per test. `jest.resetModules()` forces a clean module re-import, which is the correct approach for testing module-level lazy initialization. |

File: lib/__tests__/contentApi.test.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ describe('useContentAPI Hook', () => {
         pages = await result.current.getAllPages()
       })
 
-      expect(fetch).toHaveBeenCalledWith('http://localhost:3000/api/admin/pages', { credentials: 'include' })
+      expect(fetch).toHaveBeenCalledWith('/api/admin/pages', { credentials: 'include' })
       expect(pages).toEqual(mockPages)
```
[the remark this answers is not in the document]
Comment (frankbria) [reply lib/__tests__/contentApi.test.js]: **Rejected** — this is a false positive. The whole point of issue #42 is that admin endpoints are now local Next.js API routes backed by SQLite, not backend API calls. Using `NEXT_PUBLIC_API_BASE` for these routes would incorrectly proxy them through the backend. Relative URLs are architecturally correct here — the `fallback` rewrite pattern ensures local routes take precedence. CLAUDE.md has been updated to document that admin API calls use relative URLs while public/catalog calls continue using `NEXT_PUBLIC_API_BASE`.
